armv7a9-zynq7000/syspage.c

```diff
@@ -0,0 +1,501 @@
+
+void syspage_save(void)
+{
+	int i;
+
+	syspage_common.argCnt++; /* The last char is '\0' */
+	syspage_common.syspage->arg = (char *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	syspage_common.syspage->syspagesz += syspage_common.argCnt;
+	low_memcpy((void *)(syspage_common.syspage->arg), syspage_common.args, syspage_common.argCnt);
+
+	syspage_common.syspage->progs = (syspage_program_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	syspage_common.syspage->progssz = syspage_common.progsCnt;
+	syspage_common.syspage->syspagesz += (syspage_common.progsCnt * sizeof(syspage_program_t));
+	low_memcpy((void *)(syspage_common.syspage->progs), syspage_common.progs, syspage_common.progsCnt * sizeof(syspage_program_t));
+
+	syspage_common.syspage->maps = (syspage_map_t *)((void *)syspage_common.syspage + syspage_common.syspage->syspagesz);
+	for (i = 0; i < syspage_common.mapsCnt; ++i)
+		low_memcpy((void *)((void *)syspage_common.syspage->maps + i * sizeof(syspage_map_t)), &syspage_common.maps[i].map, sizeof(syspage_map_t));
+
+	syspage_common.syspage->syspagesz += (syspage_common.mapsCnt * sizeof(syspage_map_t));
+	syspage_common.syspage->mapssz = syspage_common.mapsCnt;
+}
+
+
+/* TODO: align text and show map's attributes as string based on integer */
+void syspage_show(void)
+{
+	int i = 0;
+	plostd_printf(ATTR_LOADER, "\nSyspage addres: 0x%p\n", syspage_getAddress());
+	plostd_printf(ATTR_NONE, "--------------------------\n");
+
+	plostd_printf(ATTR_LOADER, "Kernel sections: \n");
+	plostd_printf(ATTR_USER, ".text: %p \tsize: %p\n", syspage_common.syspage->kernel.text, syspage_common.syspage->kernel.textsz);
+	plostd_printf(ATTR_USER, ".data: %p \tsize: %p\n", syspage_common.syspage->kernel.data, syspage_common.syspage->kernel.datasz);
+	plostd_printf(ATTR_USER, ".bss:  %p \tsize: %p\n", syspage_common.syspage->kernel.bss, syspage_common.syspage->kernel.bsssz);
+
+	plostd_printf(ATTR_LOADER, "\nPrograms number: %d\n", syspage_common.progsCnt);
+	if (syspage_common.progsCnt) {
+		plostd_printf(ATTR_LOADER, "NAME\t\tSTART\t\tEND\t\tIMAP\tDMAP\n");
+		for (i = 0; i < syspage_common.progsCnt; ++i)
+			plostd_printf(ATTR_USER, "%s\t\t%p\t%p\t%d\t%d\n", syspage_common.progs[i].cmdline, syspage_common.progs[i].start, syspage_common.progs[i].end, syspage_common.progs[i].imap, syspage_common.progs[i].dmap);
+	}
+
+	plostd_printf(ATTR_LOADER, "\n");
+	plostd_printf(ATTR_LOADER, "Mulimaps number: %d\n", syspage_common.mapsCnt);
+	if (syspage_common.mapsCnt) {
+		plostd_printf(ATTR_LOADER, "ID\tNAME\tSTART\t\tEND\tTOP\t\tFREESZ\tATTR\n");
+		for (i = 0; i < syspage_common.mapsCnt; ++i)
+			plostd_printf(ATTR_USER, "%d\t%s\t%p\t%p\t%p\t%d\t%d\n",  syspage_common.maps[i].map.id,  syspage_common.maps[i].map.name, syspage_common.maps[i].map.start,
+						  syspage_common.maps[i].map.end, syspage_common.maps[i].top, syspage_common.maps[i].map.end - syspage_common.maps[i].top, syspage_common.maps[i].map.attr);
```

Comment:
These lines are too long for me ( more then 200 chars).

Reply:
This comment and several above are also associated with syspages on the other platforms. I will not make changes within this PR, it should be done globally.

Nevertheless, thank for comments and I will refer to it in the next PR where syspage's structure will be updated for Zynq-7000 and the others.